src/mpid/ch4/netmod/ofi/ofi_events.c

```diff
@@ -674,7 +687,10 @@ static int am_recv_event(struct fi_cq_tagged_entry *wc, MPIR_Request * rreq)
     if ((uo_msg = MPIDI_OFI_am_claim_unordered_msg(fi_src_addr, next_seqno)) != NULL) {
         am_hdr = &uo_msg->am_hdr;
         orig_buf = am_hdr;
-        goto repeat;
```

Comment:
Okay, it only repeats when you have additional message from unordered queue. How about name the label something like `more_message_from_unordered_queue`? If you prefer short labels, maybe just `more_message`?

Reply:
I feel `fn_repeat` is fine here; it indicates a loop and the position where we should start and possibly repeat our packet processing.

Reply:
Yes, that is obvious. It leaves the question that why do we need to repeat, when do we need to repeat, and when do we exit the repeat. These questions are very *un*-obvious until we read all the way to the end and maybe go back and forth a couple of times. I am not going to argue if you insist, but the fact that we get so easily confused without careful reading is an indicator that the code can be improved.

Reply:
I think it is impossible for us to let one label to explain why we need to repeat. Even if `more_message` is used, we can still ask, why we have more messages. The explanation is not self-contained. We still have to read the code and understand the logic before we really know why there is a repeat.

Reply:
> I think it is impossible ...

A lot of bad code can happen with this type of thinking. We can acknowledge that we don't how to do better, better never say "impossible"... 😄

Reply:
maybe I should use `difficult`